todo_app/app.py

```diff
@@ -1,36 +1,66 @@
+import json
 import os
+
+from flask.wrappers import Response
+
+from todo_app.items.item import ItemEncoder
+from todo_app.view_model import ViewModel
 from todo_app.items.trello.transport import TrelloTransport
 from flask import Flask, render_template, request
 
-from todo_app.flask_config import Config
-
 from todo_app.items.trello.items import Items
 
-app = Flask(__name__)
-app.config.from_object(Config)
-trello_transport = TrelloTransport(os.environ.get('TRELLO_API_KEY'), 
+def create_app():
+    app = Flask(__name__)
+    # We specify the full path and remove the import for this config so
+    # it loads the env variables when the app is created, rather than when this file is imported
+    app.config.from_object('todo_app.flask_config.Config')
+
+    trello_transport = TrelloTransport(os.environ.get('TRELLO_API_KEY'), 
         os.environ.get('TRELLO_SERVER_TOKEN'))
-item_registry = Items(trello_transport, os.environ.get('TRELLO_BOARD_ID'))
+    item_registry = Items(trello_transport, os.environ.get('TRELLO_BOARD_ID'))
+
+    @app.route('/')
+    def index():
+        app.logger.info("Request received to load root")
+        return render_template('index.html')
+
+    @app.route('/todo')
+    def get_all_items():
+        items = sorted(item_registry.get_items(), key=lambda item: item.status, reverse=True)
+        item_view_model = ViewModel(items)
+        status = request.args.get('status')
+        switch = {
+            "todo": item_view_model.to_do_items,
+            "doing": item_view_model.doing_items,
+            "done": item_view_model.done_items,
+        }
+
+        return json.dumps(sorted(switch.get(status, item_view_model.items), key=lambda item: item.status, reverse=True), cls=ItemEncoder)
```

Comment:
Making status optional is kind of nice, but your actual app always sets a status.
So your test for this endpoint is testing a flow that the app never uses, which isn't ideal. App code generally shouldn't exist only for use in a test.

Reply:
Yeah I think I agree. I looked up switch statements in Python and this was the recommended approach. However I'm not sure I like it as it has a few foibles. Perhaps a better approach to this is to just explicitly do the if statements and then at the end fail if the status was unknown, rather than allow it to return all? I'll look to refactor in the future.